var/spack/repos/builtin/packages/subversion/package.py

```diff
@@ -67,9 +67,9 @@ def install(self, spec, prefix):
         options.append('--with-sqlite=%s' % spec['sqlite'].prefix)
         options.append('--with-serf=%s' % spec['serf'].prefix)
 
-        if spec.satisfies('^swig'):
+        if 'swig' in spec:
             options.append('--with-swig=%s' % spec['swig'].prefix)
-        if spec.satisfies('+perl'):
```

Comment:
The `+perl` in `s.satisfies('+perl')` is testing for is the `+perl` variant on the root.  It doesn't test that against the dependencies.  For `^swig', it looks like it's not being strict enough with checking dependency relationships.